rootfs/etc/nginx/lua/test/configuration_test.lua

```diff
@@ -0,0 +1,97 @@
+_G._TEST = true
+local cjson = require("cjson")
+
+local _ngx = {
+    shared = {
+        certificate_data = {
+          get = function(self, rsc) end,
+          set = function(self, a) return true end,
+        }
+    },
+    print = function(msg) return msg end,
+    log = function(stream, msg) end,
+    var = {},
+    HTTP_OK = 200,
+    HTTP_BAD_REQUEST = 400,
+    HTTP_CREATED = 201,
+    HTTP_INTERNAL_SERVER_ERROR = 500,
+    ERR = nil,
+    req = {
```

Comment:
would not just mocking `ngx.req` be enough?
```
_G.ngx.req = {
  ...
}
```

Also please make sure you unmock it after your test finishes otherwise you're leaking a state.

Reply:
Yes, will unmock after.

Unless I'm missing something just mocking `ngx.req` wouldn't be enough.  The tests require access to `ngx.status`, which wouldn't be accessible unless `ngx` was mocked. By mocking `ngx` I had to add the other mocked functions and variables that are used by `configuration.lua`.